### DISCUSSION TITLE
Multi-Candidates election can be manipulated by submitting duplicate positions

### DISCUSSION THREAD

**Discussion by simpletrontdip** [Agoric/agoric-sdk#6711] [General]
Hi, while reading through the code, I've found out that `submitVote` did not check for the uniqueness of positions.
Since inside `countVotes`, we only find the position index and add the `shares`, so some voters can exploit this to manipulate the election's result, say:

```js
submitVote(['A', 'A', 'A'])

// instead of
submitVote(['A', 'B', 'C'])
```

https://github.com/Agoric/agoric-sdk/blob/ca0b874d781294d193f9d185365f47b284f55c0a/packages/governance/src/multiCandidateVoteCounter.js#L68-L76

https://github.com/Agoric/agoric-sdk/blob/ca0b874d781294d193f9d185365f47b284f55c0a/packages/governance/src/multiCandidateVoteCounter.js#L168-L183

**Comment by simpletrontdip** [comment]
Something like this may help
```js
        const chosenCount = chosenPositions.length;
        const uniqueCount = new Set(chosenPositions).size;

        chosenCount === uniqueCount ||
          Fail`Duplicated positions found`;
```

**Comment by Chris-Hibbert** [comment]
I created an issue #6721.  Thanks for reporting this!
